src/v/kafka/requests/consumer_records.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "bytes/iobuf.h"
+#include "kafka/types.h"
+#include "model/fundamental.h"
+#include "model/record_batch_reader.h"
+
+#include <optional>
+
+namespace kafka {
+
+///\brief consumer_records are a concatenation of multiple model::record_batch
+/// on the wire; without a size header (c.f. a kafka array<thing>)
+class consumer_records {
```

Comment:
i'd also make this a `class consumer_records final : public record_batch_reader::impl {...} `